packages/kit/src/build.ts

```diff
@@ -1,135 +1,168 @@
 import type { Configuration as WebpackConfig, WebpackPluginInstance } from 'webpack'
 import type { UserConfig as ViteConfig, Plugin as VitePlugin } from 'vite'
 import { useNuxt } from './context'
+import { toArray } from './utils'
 
 export interface ExtendConfigOptions {
   /**
    * Install plugin on dev
    * @default true
    */
   dev?: boolean
+
   /**
    * Install plugin on build
    * @default true
    */
   build?: boolean
+
   /**
    * Install plugin on server side
    * @default true
    */
   server?: boolean
+
   /**
    * Install plugin on client side
    * @default true
    */
   client?: boolean
+
   /**
    * Prepends the plugin to the array with `unshift()` instead of `push()`.
    */
   prepend?: boolean
 }
 
-export interface ExtendWebpackConfigOptions extends ExtendConfigOptions {}
+export interface ExtendWebpackConfigOptions extends ExtendConfigOptions { }
 
-export interface ExtendViteConfigOptions extends ExtendConfigOptions {}
+export interface ExtendViteConfigOptions extends ExtendConfigOptions { }
 
 /**
- * Extend webpack config
- *
- * The fallback function might be called multiple times
- * when applying to both client and server builds.
+ * Extends the webpack configuration. Callback function can be called multiple times, when applying to both client and server builds.
+ * @param callbackFunction - A callback function that will be called with the webpack configuration object.
+ * @param options - Options to pass to the callback function. See {@link https://nuxt.com/docs/api/kit/builder#options available options}.
+ * @see {@link https://nuxt.com/docs/api/kit/builder#extendwebpackconfig documentation}
  */
 export function extendWebpackConfig (
-  fn: ((config: WebpackConfig) => void),
+  callbackFunction: ((config: WebpackConfig) => void),
   options: ExtendWebpackConfigOptions = {}
 ) {
   const nuxt = useNuxt()
 
-  if (options.dev === false && nuxt.options.dev) {
-    return
-  }
-  if (options.build === false && nuxt.options.build) {
+  if (
+    (options.dev === false && nuxt.options.dev)
+    || (options.build === false && !nuxt.options.dev)
+  ) {
```

Comment:
I think this is clearer:


```suggestion
  if (options.dev === false && nuxt.options.dev) {
    return
  }
  if (options.build === false && nuxt.options.build) {
```